core/server/src/test/java/alluxio/master/file/FileSystemMasterTest.java

```diff
@@ -1414,6 +1414,72 @@ public void sameDstConcurrentRename() throws Exception {
   }
 
   /**
+   * Tests renaming files concurrently to the same destination from different sources will
+   * succeed only once.
+   */
+  @Test
+  public void sameDstDifferentSrcConcurrentRename() throws Exception {
+    int numThreads = 10;
+    final AlluxioURI[] srcs = new AlluxioURI[numThreads];
+    final AlluxioURI[] dsts = new AlluxioURI[numThreads];
+    AlluxioURI dir1 = new AlluxioURI("/dir1");
+    AlluxioURI dir2 = new AlluxioURI("/dir2");
+    mFileSystemMaster.createDirectory(dir1, CreateDirectoryOptions.defaults());
+    mFileSystemMaster.createDirectory(dir2, CreateDirectoryOptions.defaults());
+    for (int i = 0; i < numThreads; i++) {
+      if (i % 2 == 0) {
+        srcs[i] = dir1.join("file" + i);
```

Comment:
can we make type 1 source like `"file1_" + i`, type 2 source like `"file1_" + i`,
and assert `file.getName().startsWith("file1_")` in `/dir1` and `file.getName().startsWith("file2_")` in `/dir2`?